src/qt/clientmodel.cpp

```diff
@@ -242,8 +242,9 @@ static void BlockTipChanged(ClientModel *clientmodel, bool initialSync, int heig
         clientmodel->cachedBestHeaderHeight = height;
         clientmodel->cachedBestHeaderTime = blockTime;
     }
-    // if we are in-sync or if we notify a header update, update the UI regardless of last update time
-    if (fHeader || !initialSync || now - nLastUpdateNotification > MODEL_UPDATE_DELAY) {
+
+    // During initial sync, block notifications, and header notifications from reindexing are both throttled.
```

Comment:
I find this very confusing, and even if I look past the grammar error, probably outright wrong since header notifications are only throttled for reindex, not initial sync?

Reply:
IIUC, `fReindex==true` implies `initialSync==true`: https://github.com/bitcoin/bitcoin/blob/2bdc476d4d23256d8396bb9051a511f540d87392/src/validation.cpp#L1281-L1282

Anyway, mind suggesting a better wording?

Reply:
> I find this very confusing, and even if I look past the grammar error, probably outright wrong since header notifications are only throttled for reindex, not initial sync?

This is pretty confusing, and I was probably trying to pack too much information in a short comment. There should be chances to clean this up in the future with followups from #18136 or https://github.com/bitcoin/bitcoin/pull/18121#pullrequestreview-357730284

Ideally the code would just be more direct and not need a comment stating what it does:

```c++
enum NotificationType { BLOCK_TIP_CHANGED, HEADER_TIP_CHANGED };
enum NotificationStatus { INIT_REINDEX, INIT_DOWNLOAD, POST_INIT };

void HandleNotifications(NotificationType notification, NotificationStatus status) {
   bool throttle = status == INIT_REINDEX || (status == INIT_DOWNLOAD && notification == BLOCK_TIP_CHANGED);
   if (!throttle || now - nLastUpdateNotification > MODEL_UPDATE_DELAY) ...
```

Reply:
Addressed in #18152